src/status_im2/common/standard_authentication/enter_password/view.cljs

```diff
@@ -0,0 +1,47 @@
+(ns status-im2.common.standard-authentication.enter-password.view
+  (:require [utils.i18n :as i18n]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [utils.re-frame :as rf]
+            [status-im2.common.standard-authentication.enter-password.style :as style]
+            [status-im2.common.standard-authentication.password-input.view :as password-input]))
+
+(defn view
+  [{:keys [on-enter-password button-label]}]
+  (let [{:keys [key-uid display-name
+                customization-color]}       (rf/sub [:profile/multiaccount])
+        {:keys [error processing password]} (rf/sub [:profile/login])
+        sign-in-enabled?                    (rf/sub [:sign-in-enabled?])
+        profile-picture                     (rf/sub [:profile/login-profiles-picture key-uid])]
+    [:<>
+     [rn/view {:style style/enter-password-container}
+      [rn/view
+       [quo/text
+        {:accessibility-label :sync-code-generated
+         :weight              :bold
+         :size                :heading-1
+         :style               {:margin-bottom 4}}
+        (i18n/label :t/enter-password)]
+       [rn/view
+        {:style style/context-tag}
+        [quo/context-tag {} profile-picture display-name]]
+       [password-input/view
+        {:processing       processing
+         :error            error
+         :default-password password
+         :sign-in-enabled? sign-in-enabled?}]
+       [rn/view {:style style/enter-password-button}
+        [quo/button
+         {:size                40
+          :type                :primary
+          :customization-color (or customization-color :primary)
+          :accessibility-label :login-button
+          :icon-left           :i/reveal
+          :disabled?           (or (not sign-in-enabled?) processing)
+          :on-press            (fn []
+                                 (rf/dispatch [:set-in [:profile/login :key-uid] key-uid])
```

Comment:
Just a story...

I've seen this `set-in` pattern in more than one re-frame project as a way of helping *reduce re-frame boilerplate*. Numerous times I've seen it causing UI glitches because the UI would dispatch multiple `set-in` events in succession and they would end up in different rendering batches in the re-frame queue. The more reliable solution is to mutate the app db alongside other effects in the same event handler, like an atomic operation.

The re-frame docs explicitly tell us re-frame can't guarantee the order of events between different dispatches, so this is also another problem of using this `set-in` pattern, some devs will incorrectly assume the setter happens first.

The `set-in` pattern also tends to couple the UI with the app db shape because the UI now needs to know how the data is structured behind the scenes. Some re-frame projects even go as far as using a `get-in` subscription, as yet again, a way to make accessing the app db more convenient from the UI. And things don't end there, some projects create other "wrapper events" around the core app db atom, like `assoc-in`, `update-in`, and so on. All bad for maintainability in my experience.

End of story :)

FYI: @ulisesmac, @cammellos, @flexsurfer, @erikseppanen, @smohamedjavid, @J-Son89

Reply:
Thanks @ilmotta, if there is some guidance on what is best to do I will be happy to implement. :)

Reply:
At this stage I just wanted to raise awareness @J-Son89 and I agree with you, it's not something for your PR alone to change or decide.

The following is a known post about some patterns seen in re-frame apps. Note that some ideas existed in redux apps before re-frame was a thing, so maybe finding guidance on redux can be more fruitful.

https://vvvvalvalval.github.io/posts/some-re-frame-patterns-for-composability.html#generic_subscriptions_and_events_for_app-db_paths

Reply:
yes set-in is legacy code, we event have TODO
```clojure
;;TODO :replace by named events
(rf/defn set-event
  {:events [:set]}
  [{:keys [db]} k v]
  {:db (assoc db k v)})

;;TODO :replace by named events
(rf/defn set-once-event
  {:events [:set-once]}
  [{:keys [db]} k v]
  (when-not (get db k)
    {:db (assoc db k v)}))

;;TODO :replace by named events
(rf/defn set-in-event
  {:events [:set-in]}
  [{:keys [db]} path v]
  {:db (assoc-in db path v)})
  ```
  but it should be more visible i guess, so devs won't use it